### PR TITLE
Xcode style indentation for guards and enums

### PR DESCRIPTION
**Pull request by tremblay**
setting Xcode indentation to enabled will cause guards to format like
```
guard let foo = foo
    else {
        return
}
```
or
```
guard let foo = foo
    else { return }
```
and will cause enums to format like
```
enum Foo {
    case foo,
    bar
}
```

Both of these match the way Xcode will indent by default, which may be preferable for teams that have been using Xcode indentation on a large code base prior to using SwiftFormat!

### CODE REVIEW

File: Sources/Rules.swift [hunk 1]
```diff
@@ -867,8 +867,8 @@ public struct _FormatRules {
                 indentStack.removeLast()
                 indentCounts.removeLast()
                 linewrapStack.removeLast()
-                scopeStartLineIndexes.removeLast()
-                scopeStack.removeLast()
+                _ = scopeStartLineIndexes.popLast()
```
Comment (nicklockwood) [inline Sources/Rules.swift]: I assume this is due to the extra popScope() added below, but it seems a bit fragile. Won't this result in the wrong scope getting popped if the guard scope wasn't at the bottom of the stack?

**Comment by nicklockwood** [comment]
This looks pretty great, thanks!

File: Sources/Rules.swift (hunk 1 above)
Comment (tremblay) [reply Sources/Rules.swift]: You are correct, and this was definitely too fragile. I'm not 100% certain that the adjusted code is correct either, but it seems to work. Basically, I haven't fully wrapped my head around all the stacks and counts, so I'm not totally sure that I'm setting them back correctly...

**Comment by tremblay** [comment]
@nicklockwood Any thoughts about that solution for the scope pop?

**Comment by nicklockwood** [comment]
@tremblay I think this has the same problem. Those various stacks are supposed to be in sync and so it seems risky to pop just a subset of them. I'll need to review it carefully and come up with some test cases in order to be confident it's correct, and won't crash under certain conditions.

**Comment by tremblay** [comment]
@nicklockwood I see what you mean. I think it's working because my fix basically just treats the else clause as `linewrapped` without modifying the scope stack. Which could mean I've fixed it wrong or could mean I'm just confused? I'm going to test it out some more locally, and don't hesitate to reach out if you want me to make any changes or help you test!

**Comment by tremblay** [comment]
I think my logic for when to pop is now matching what we would expect with one substantial issue. Basically the goal I avoid calling `indentStack.removeLast()` on line 881, and then perform the same work that would have happened there on line 1010.

I think there's a remaining bug with the way I'm checking for the `guard` keyword, it seems to be a little over-zealous about finding guards above the current scope. The same bug applies to my `formatter.index(of: .keyword("case"), before: i) == nil,` change I think. I'll keep digging to see if I can find a better solution for that piece of it.

**Comment by nicklockwood** [comment]
@tremblay thanks for spending the time on this. It's looking pretty close to ready - I'll do some more testing and then ship it in the next release if I don't find any problems.

**Comment by tremblay** [comment]
@nicklockwood I've been running it against the code in the Snapshots folder and there are a few more issues I think.
```
guard let polygon = Polygon([
        Vertex(Vector(-1, 1), normal),
    Vertex(Vector(-1, 1), normal),
```
for example

and also there may be a bug with the enum indentation where multi-line if statements that start with `if case let` end up under-indented.

I should have time to work on these tomorrow, I'll plan to open a PR to develop with a fix if that makes the most sense?

**Comment by nicklockwood** [comment]
@tremblay sure - I can merge any further changes you make into your original PR. I didn't get a chance to look at this today, but I'm happy to take over if you get stuck.

**Pull request merged**